packages/flutter_tools/lib/src/base/user_messages.dart

```diff
@@ -193,6 +193,11 @@ class UserMessages {
   String get visualStudioMissing =>
       'Visual Studio not installed; this is necessary for Windows development.\n'
       'Download at https://visualstudio.microsoft.com/downloads/.';
+  String get visualStudioIsPrerelease => 'The current Visual Studio installation is a pre-release version';
+  String get visualStudioNotLaunchable =>
+      'The current Visual Studio installation is not launchable. Please reinstall Visual Studio.';
+  String get visualStudioIsIncomplete => 'The current Visual Studio installation is incomplete. Please reinstall Visual Studio';
+  String get visualStudioRebootRequired => 'Visual Studio requires a reboot of your system to complete installation';
```

Comment:
Please give these strings trailing `.`s for consistency with the other messages.

Reply:
Done